processing/src/main/java/io/druid/query/dimension/ListFilteredDimensionSpec.java

```diff
@@ -110,7 +110,7 @@ public IndexedInts getRow()
       @Override
       public int getValueCardinality()
       {
-        return matched.size();
+        return selector.getValueCardinality();
```

Comment:
this breaks contract of getValueCardinality(), this should return cardinality "after" filtering.